app/src/main/java/org/thoughtcrime/securesms/home/HomeViewModel.kt

```diff
@@ -205,6 +214,19 @@ class HomeViewModel @Inject constructor(
 
     fun getCurrentUsername() = usernameUtils.getCurrentUsernameWithAccountIdFallback()
 
+    fun blockContact(accountId: String) {
+        viewModelScope.launch {
+            val recipient = Recipient.from(context, Address.fromSerialized(accountId), false)
+            storage.setBlocked(listOf(recipient), isBlocked = true)
+        }
```

Comment:
Do we need to launch these in a coroutine? It seems like there's no suspend function inside and running them on main thread doesn't make much sense neither

Reply:
Or maybe you were meant to do this?

```suggestion
        viewModelScope.launch(Dispatchers.Default) {
            val recipient = Recipient.from(context, Address.fromSerialized(accountId), false)
            storage.setBlocked(listOf(recipient), isBlocked = true)
        }
```

Same for the `deleteContact` method

Reply:
Yep, I forgot the default on the VM was main and not default...